### PR TITLE
Fix filter

### PR DESCRIPTION
**Pull request by Imafikus**

### CODE REVIEW

File: local_runner/python/datastore.py [hunk 1]
```diff
@@ -101,12 +101,22 @@ def fetch(self, limit=None, offset=None):
             result = result[:limit]
 
         return result
+    
+    def cmp(self, a, b):
+        return (a > b) - (a < b)
```
Comment (profMagija) [inline local_runner/python/datastore.py]: what is this ?

File: local_runner/python/datastore.py [hunk 2]
```diff
@@ -101,12 +101,22 @@ def fetch(self, limit=None, offset=None):
...
     def _make_orderer(self):
-        return lambda ent: tuple(
-            (-1 if o[0] == "-" else 1) * getattr(ent, o.strip("+-"), 0)
-            for o in self.order
-        )
+        def order_func(item):
+            sort_keys = []
+            for order_key in self.order:
+                desc = order_key.startswith('-')
+                key = order_key[1:] if desc else order_key
```
Comment (profMagija) [inline local_runner/python/datastore.py:111]: this does not work for `"+x"` keys

File: local_runner/python/datastore.py [hunk 3]
```diff
@@ -101,12 +101,22 @@ def fetch(self, limit=None, offset=None):
...
+                value = item.get(key, 0)  # Assuming numerical values, defaulting to 0
```
Comment (profMagija) [inline local_runner/python/datastore.py:112]: you sure `get` works ?

File: local_runner/python/datastore.py (hunk 1 above)
Comment (Imafikus) [reply local_runner/python/datastore.py]: Ooops

File: local_runner/python/datastore.py (hunk 3 above)
Comment (Imafikus) [reply local_runner/python/datastore.py:112]: We tested it locally, it looks like it's working

File: local_runner/python/datastore.py (hunk 2 above)
Comment (Imafikus) [reply local_runner/python/datastore.py:111]: According to the docs, you only use `-`, ascending keys are without `+`.

Link: https://cloud.google.com/datastore/docs/concepts/queries#sort_orders

**Review by profMagija — APPROVED** [review]

**Pull request merged**